File: 0443-string-compression/0443-string-compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        int n = chars.size();
        if (n <= 1) return n;
        
        // int index = 0;
        // int i = 0;
        
        // while (i < n) {
        //     char currentChar = chars[i];
        //     int count = 0;
            
        //     // Count the number of occurrences of the current character
        //     while (i < n && chars[i] == currentChar) {
        //         i++;
        //         count++;
        //     }
            
        //     // Write the character to the result
        //     chars[index++] = currentChar;
            
        //     // If count is greater than 1, write the count to the result
        //     if (count > 1) {
        //         string countStr = to_string(count);
        //         for (char c : countStr) {
        //             chars[index++] = c;
        //         }
        //     }
        // }
        
        // return index;
        int i=0;
        int index=0;
        while(i<n){
            char curr=chars[i];
            int count=0;
            while(i<n && curr==chars[i]){
                count++;
                i++;
            }
            chars[index++]=curr;
            if(count>1){
                string str=to_string(count);
                for(int i=0;i<str.size();i++){
                    chars[index++]=str[i];
                }

            }

        }
        return index;
    }
};

File: 0443-string-compression/0443-string-compression.py
from typing import List

class Solution:
    def compress(self, chars: List[str]) -> int:
        j = 0  
        i = 0  
        
        while i < len(chars):
            x = chars[i]
            c = 0
            
            while i < len(chars) and chars[i] == x:
                c += 1
                i += 1
            
            chars[j] = x
            j += 1
            
            if c > 1:
                for digit in str(c):
                    chars[j] = digit
                    j += 1
        
        return j
